docs/usage.rst

```diff
@@ -189,6 +189,62 @@ Since version ``0.20.0``, the ``pika-bind`` helper script is bundled with pika.
 and then runs the given command. ``pika-bind`` is a more convenient alternative to manually setting ``PIKA_PROCESS_MASK``
 when pika is used together with a runtime that may reset the process mask of the main thread, like OpenMP.
 
+.. _logging:
+
+Logging
+=======
+
+The pika runtime logs warnings and more severe messages to stderr by default, using `spdlog
+<https://github.com/gabime/spdlog>`_. To change the logging level, set the ``PIKA_LOG_LEVEL``
+environment variable to a value between 0 (trace) and 6 (off) (the values correspond to levels in
+spdlog). The log messages are sent to stderr by default. The destination can be changed by setting
+the ``PIKA_LOG_DESTINATION`` environment variable. Supported values are:
+
+- ``cerr``
+- ``cout``
+- any other value is interpreted as a path to a file
+
+pika will by default print messages in the following format:
+
+.. code-block::
+
+   [2024-04-18 13:45:07.095279283] [pika] [info] [host:machine(-1)] [pid:2786603] [tid:2786607] [pool:0000/0003/0003] [parent:----/----] [task:0x7fa6a4077cf0/pika_main] [set_thread_state.cpp:205:set_thread_state] set_thread_state: thread(0x7fa6a802c8d0), description(<unknown>), new state(pending), old state(suspended)
+
+The fields are as follows:
+
+- ``[2024-04-18 13:45:07.095279283]``: The timestamp of the message.
+- ``[pika]``: An identifier present in all pika's logs.
+- ``[info]``: The severity level of the message.
+- ``[host:machine(-1)]``: The hostname and the MPI rank of the process (``-1`` if MPI is disabled).
+- ``[pid:2786603]``: The process id as reported by the operating system.
+- ``[tid:2786607]``: The thread id as reported by the operating system.
+- ``[pool:0000/0003/0003]``: The pika thread pool and worker thread ids: the first component is the
+  thread pool id, the second is the global worker thread id (unique across all thread pools), and
+  the third is the local worker thread id (unique only within the current thread pool). If the
+  message is not logged from a thread on the pika runtime you may see ``[pool:----/----/----]``. You
+  may also see ``[pool:----/0004/----]``. This corresponds to the main thread of the application,
+  which does not belong to a thread pool, but still has a global worker thread id.
```

Comment:
Some of the entries, like this one, are quite verbose. I'm wondering if it would be beneficial to have simply
```
- ``[pool:0000/0003/0003]``: The pika thread pool and worker thread ids: the first component is the
  thread pool id, the second is the global worker thread id (unique across all thread pools), and
  the third is the local worker thread id (unique only within the current thread pool).
```
And move the details in dedicated per-item subsections. Same thing for some of the entries below.

Reply:
That's a good point. I was worried that the special log entries (like `----/0004/----`) would be too confusing without further explanation so I opted for being more verbose. Is your suggestion to structure it a bit more like this (in pseudo-markdown):
```
# Logging

Intro to logging.

Description of fields:
- list
- of
- short
- descriptions

## Thread pool and worker threads

Longer description of the possible values.

## Task id and description

Longer description of the possible values.
```
? Or perhaps:
```
# Logging

Intro to logging.

Description of fields:
- list
- of
- short
- descriptions

Longer description of the possible values of thread pool and worker thread ids.

Longer description of the possible values of task ids and descriptions.
```
?

Since not all fields need further explanation I thought it makes sense to keep it straightforward with all the information in the list. Seeing the above options I think perhaps the second could be a bit nicer to read. Did you have something else in mind? What do you think?

Reply:
I was suggesting the first option, but both are equivalent IMO. I was just concerned that having very long bullet points might be difficult to read/follow.

Reply:
I've pushed a small restructuring in https://github.com/pika-org/pika/pull/1093/commits/bfd61c516b1f8607969bd45bacde5f186b0d6e8e. Rendered here for convenience:
![2024-04-19-141838_792x1226_scrot](https://github.com/pika-org/pika/assets/42977/aaff07cb-8baa-4c06-bbe6-fba713affb60)
Does that look any better to you?